format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSinkConfig.java

```diff
@@ -84,13 +84,32 @@ public void validate(FailureCollector collector) {
           .withConfigProperty(NAME_SUFFIX).withStacktrace(e.getStackTrace());
       }
     }
+
+    FileFormat format = FileFormat.CSV;
     try {
-      getFormat();
+      format = getFormat();
     } catch (IllegalArgumentException e) {
       collector.addFailure(e.getMessage(), null).withConfigProperty(NAME_FORMAT).withStacktrace(e.getStackTrace());
     }
     try {
-      getSchema();
+      Schema schema = getSchema();
+      // Checks if the output schema has fields that are not simple type.
+      // If there are non-simple field types, then serializing as CSV might not be appropriate.
+      // This restriction is on Csv, Delimited and Tab separated files.
+      if (format == FileFormat.CSV || format == FileFormat.DELIMITED || format == FileFormat.TSV) {
+        boolean simpleFields = true;
+        for (Schema.Field field : schema.getFields()) {
```

Comment:
Can be just

```
boolean allSimple = schema.getFields().stream().map(Schema.Field::getSchema).allMatch(Schema::isSimpleOrNullableSimple)
```

Reply:
this will logic will not short circuit right and go through all the elements.

Reply:
Found that it does. changing it.